### PR TITLE
Setup Github Actions

### PR DESCRIPTION
**Pull request by hivtushok**
This PR includes:

- setup CI/CD process;
- bump to v1.4.0.

### CODE REVIEW

File: .github/workflows/gradle-build.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+name: Build & test with Gradle
+
+env:
+  SONATYPE_USER: ${{ secrets.SONATYPE_USER }}
+  SONATYPE_PASSWORD: ${{ secrets.SONATYPE_PASSWORD }}
+  PGP_SIGNING_KEY: ${{ secrets.PGP_SIGNING_KEY }}
+  PGP_SIGNING_PASSWORD: ${{ secrets.PGP_SIGNING_PASSWORD }}
+
+on:
+  push:
```
Comment (mvlabat) [inline .github/workflows/gradle-build.yml:13]: I think we want the following settings:

```suggestion
on:
  push:
    branches: [master]
  pull_request:
    branches: [master]
```

Comment (aquiladev) [reply .github/workflows/gradle-build.yml:11]: it is right config(suggested)!

Comment (hivtushok) [reply .github/workflows/gradle-build.yml:11]: Updated!

File: .github/workflows/gradle-build.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      - name: Cache Gradle packages
+        uses: actions/cache@v2
+        with:
+          path: ~/.gradle/caches
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
+          restore-keys: ${{ runner.os }}-gradle
+      - name: Grant execute permission for gradlew
+        run: chmod +x gradlew
+      - name: Build & test with Gradle
+        run: ./gradlew -PnexusUsername=nexusUsername -PnexusPassword=nexusPassword build
```
Comment (aquiladev) [inline .github/workflows/gradle-build.yml:34]: it looks like there is should be run of tests, but I don't see it

Comment (hivtushok) [reply .github/workflows/gradle-build.yml:34]: `gradle build` command implies tests run.

File: .github/workflows/gradle-publish.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# This workflow will build a Java project with Gradle
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-gradle
+
+name: Publish package to the Maven Central Repository with Gradle
+
+env:
+  nexusUsername: ${{ secrets.SONATYPE_USER }}
```
Comment (aquiladev) [inline .github/workflows/gradle-publish.yml:7]: why not `NEXUS_USER`?

Comment (hivtushok) [reply .github/workflows/gradle-publish.yml:7]: Here we use Sonatype account for uploads to Nexus repository manager.

Comment (hivtushok) [reply .github/workflows/gradle-publish.yml]: Renamed to eliminate the confusion.

**Comment by hivtushok** [comment]
@mvlabat @aquiladev are you good with the listed changes now?

**Review by aquiladev — APPROVED** [review]

**Review by mvlabat — APPROVED** [review]
Can't validate much of Java, but the CI script looks fine to me 👍

File: .github/workflows/gradle-publish.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+env:
+  nexusUsername: ${{ secrets.NEXUS_USER }}
+  nexusPassword: ${{ secrets.NEXUS_PASSWORD }}
+  PGP_SIGNING_KEY: ${{ secrets.PGP_SIGNING_KEY }}
+  PGP_SIGNING_PASSWORD: ${{ secrets.PGP_SIGNING_PASSWORD }}
+
+on:
+  release:
+    types: [ created ]
```
Comment (mvlabat) [inline .github/workflows/gradle-publish.yml:14]: Minor detail: do we want to do it on `created` or `published` event?

Comment (mvlabat) [reply .github/workflows/gradle-publish.yml:14]: We might want to do it on `created` event if we want to create a draft release in GitHub and test the release published to Maven. I guess the question is then: can we remove it from Maven if we don't like it? If not, then this use-case is probably irrelevant, and we just want to trigger the release process on publishing.
What do you think?

**Pull request merged**